rest/route/annotations_test.go

```diff
@@ -376,17 +376,18 @@ func TestAnnotationByTaskPutHandlerParse(t *testing.T) {
 	execution0 := 0
 	execution1 := 1
 	a := &model.APITaskAnnotation{
-		Id:     utility.ToStringPtr("1"),
-		TaskId: utility.ToStringPtr("TaskFailedId"),
-		Note:   &model.APINote{Message: utility.ToStringPtr("task-1-note_0")},
+		Id:            utility.ToStringPtr("1"),
+		TaskId:        utility.ToStringPtr("TaskFailedId"),
+		Note:          &model.APINote{Message: utility.ToStringPtr("task-1-note_0")},
+		TaskExecution: utility.ToIntPtr(1),
```

Comment:
Can we please also add a test where an annotation doesn't have an execution and make sure that it errors?